assets/app/view/game/spreadsheet.rb

```diff
@@ -183,40 +191,51 @@ def render_corporations
         current_round = @game.turn_round_num
 
         sorted_corporations.map.with_index do |corp_array, index|
-          render_corporation(corp_array[1], corp_array[0], current_round, index)
+          render_corporation(*corp_array, current_round, index)
         end
       end
 
       def sorted_corporations
         floated_corporations = @game.round.entities
 
-        result = @game.all_corporations.map do |c|
+        result = @game.all_corporations.reject(&:closed?).select { |c| c.minor? || c.ipoed }
+        result = result.sort.each.with_index.map do |c, order|
           operating_order = (floated_corporations.find_index(c) || -1) + 1
-          [operating_order, c]
+          [c, operating_order, order + 1]
         end
 
-        result.sort_by! do |operating_order, corporation|
-          case @spreadsheet_sort_by
-          when :cash
-            corporation.cash
-          when :id
-            corporation.id
-          when :order
-            operating_order
-          when :par_price
-            corporation.par_price&.price || 0
-          when :share_price
-            corporation.share_price&.price || 0
-          else
-            @game.player_by_id(@spreadsheet_sort_by)&.num_shares_of(corporation)
-          end
+        result = result.map { |c, _, order| [c, order, order] } if result.to_enum.with_object(1).map(&:[]).all?(0)
```

Comment:
what is this?

Reply:
If corporations are equal in the attribute we are sorting by, it reverts to sorting by market order. During the OR we have an array explicitly storing the order, so we use it. When we don't have that array, we use the current order in the market.

Reply:
i feel like you could do this easier, make it more clean

Reply:
The problem is that order can mean 2 different things and sometimes we care about both of those at the same time.

I changed it the little that I could. I hope it's good enough